crypto/dh/dh_kdf.c

```diff
@@ -185,3 +190,4 @@ int DH_KDF_X9_42(unsigned char *out, size_t outlen,
     EVP_MD_CTX_cleanup(&mctx);
     return rv;
 }
+#endif
```

Comment:
@levitte I can reproduce the empty translation unit on the dh_kdf.c if I include e_os.h instead of e_os2.h as I've seen in other files (ebcdic.c for example). The -pedantic flag added manually or by --strict-warnings does not trigger a failure on the current patch.

Using e_os.h instead of e_os2.h in ebcdic.c makes the dummy thing unnecessary there as well.
Apparently e_os.h ends up inserting lots of typedefs and extern declarations that make the translation unit not empty. With e_os2.h the macroexpanded C file is just empty, thus the error.